### PR TITLE
test against pandas rc

### PR DESCRIPTION
**Pull request by TomAugspurger**
I messed up #2813, sorry for the noise.

### CODE REVIEW

**Comment by TomAugspurger** [comment]
The failures:

- `.assign(a=scalar). This has been fixed between the RC and pandas master, (https://github.com/pandas-dev/pandas/pull/17902 I think(
  * https://travis-ci.org/dask/dask/jobs/292048042#L5314
  * https://travis-ci.org/dask/dask/jobs/292048042#L4900 
- `str(CategoricalDtype)`. pandas changed this back to 'category' to help with projects that relied on that. This is a test-only failure, just have to remove the `if PANDAS_VERSION >= 0.21.0`.
  * https://travis-ci.org/dask/dask/jobs/292048042#L5064
  * https://travis-ci.org/dask/dask/jobs/292048042#L5229
  * https://travis-ci.org/dask/dask/jobs/292048042#L5282
  * https://travis-ci.org/dask/dask/jobs/292048042#L6618
- `.rename(index=., columns=.)` or `.rename(mapper, axis=.)`. In https://github.com/pandas-dev/pandas/pull/17800 I missed a case that's ambiguous, though I may have a way to handle it. Doing that now. Either way, we'll update dask to use named arguments.

I'll also setup an allowed failure that tests against pandas nightly. If it causes issues later on we'll revert that.

**Comment by TomAugspurger** [comment]
https://github.com/pandas-dev/pandas/pull/17966 for the change on pandas side. Assuming that's merged, dask will be OK, but `dataframe.rename` will emit a warning until the changes in https://github.com/dask/dask/pull/2814/commits/ca1456ba89be2e7577d5bee0b113ca5e05caa5eb are merged.

**Comment by TomAugspurger** [comment]
Does anyone have objections to testing against pandas nightly builds? I can revert that change in https://github.com/dask/dask/pull/2814/commits/0c4d4eb5bfadabc7445336723a5506e1eecd852b if people worry about things breaking too often.

**Comment by mrocklin** [comment]
I have no objection to trying this out.  We can revert it later if it adds significant friction.

**Comment by TomAugspurger** [comment]
OK. This should be good to go then.

**Comment by mrocklin** [comment]
@jcrist any objections?

File: continuous_integration/travis/install.sh [hunk 1]
```diff
@@ -50,6 +52,14 @@ conda install -q -c conda-forge \
     sqlalchemy \
     toolz
 
+# Install pandas dev
+conda uninstall -y --force numpy pandas
+
+if [[ ${PANDAS_DEV} ]]; then
+    PRE_WHEELS="https://7933911d6844c6c53a7d-47bd50c35cd79bd838daf386af554a83.ssl.cf2.rackcdn.com"
+    pip install --pre --no-deps --upgrade --timeout=60 -f $PRE_WHEELS numpy pandas
```
Comment (jcrist) [inline continuous_integration/travis/install.sh]: What version of numpy does this install? Master? If so I'd switch the flag from `PANDAS_DEV` to something more indicative that numpy is dev too.

**Comment by jcrist** [comment]
Seems fine to me, as long as the version of pandas/numpy installed is clear in the travis logs.

File: continuous_integration/travis/install.sh (hunk 1 above)
Comment (TomAugspurger) [reply continuous_integration/travis/install.sh]: Ah, yes I meant to ask about that. It'll be NumPy dev as well. It doesn't need to be, but I figured we may as well.

Comment (jcrist) [reply continuous_integration/travis/install.sh]: Numpy dev is fine, I'd just change the flag/comments.

**Comment by jcrist** [comment]
Looks good to me. Thanks for taking care of this @TomAugspurger.

**Pull request merged**